doc/overview.md

```diff
@@ -0,0 +1,76 @@
+`polymer-analyzer` is a static code analyzer for the web. It reads code and understands features like imports, classes, mixins, and custom element declarations.
+
+Use cases:
+ * documentation generation
+ * linting
+ * text editor and IDE plugins
+ * transforming code for bundling and sharding
+ * transforming code to update to new APIs
+
+## Goals
+
+ * Extensible: feature recognition is broken out into independent scanners. Teaching the analyzer to recognize new features is as easy as adding a new scanner.
+
+ * Fast: the analyzer walks each document as few times as possible. We use dependency-aware caching to do the minimum amount of work when a file changes.
+
+ * Standards-based: the analyzer ships with a set of scanners that are able to parse and analyze standards-based web sites.
+
+ * Loading agnostic: the user specifies both how to load files, and how URLs map into paths for the loader. This enables use cases like loading files over a network, as well as cases where the filesystem does not map 1:1 onto the url space as the source code sees it.
+
+## Design
+
+### Phases
+
+From a high level, the analyzer's work is broken up into phases.
+
+    Loading ➡ Parsing ➡ Scanning ➡ Resolving
+
+*Loading* takes a URL and returns a string representing the file's contents.
+
+*Parsing* takes that string and returns an abstract syntax tree that represents the raw syntax of the code.
+
+*Scanning* takes that AST and returns higher-level, more semantically meaningful, file-local features. While scanning, we may encounter inline documents and imports, which will cause them to be loaded, parsed, and scanned as well. A document isn't finished being scanned until all of its dependencies are also scanned.
+
+*Resolving* takes a scanned feature and all resolved features found up until that point and returns a resolved feature. Resolving can use information defined earlier in the file, or in files imported earlier in the file. It's the only point in the analysis process that can use cross-file information.
```

Comment:
I really like using "scanned" to highlight how this connects to the previous step. Can you do this on the others:
- `*Parsing* takes the loaded file contents and...`
- `*Scanning* takes the parsed AST and returns...`